static/src/javascripts/projects/common/modules/commercial/dfp/render-advert.js

```diff
@@ -128,14 +130,25 @@ define([
         removePlaceholders(advert.node);
 
         return applyCreativeTemplate(advert.node).then(function (isRendered) {
-            renderAdvertLabel(advert.node);
+            return renderAdvertLabel(advert.node)
+                .then(callSizeCallback)
+                .then(addRenderedClass)
+                .then(function () {
+                    return isRendered;
+                });
 
-            var size = slotRenderEvent.size.join(',');
-            if (sizeCallbacks[size]) {
-                sizeCallbacks[size](slotRenderEvent, advert);
+            function callSizeCallback() {
+                var size = slotRenderEvent.size.join(',');
+                if (sizeCallbacks[size]) {
+                    sizeCallbacks[size](slotRenderEvent, advert);
+                }
             }
 
-            return isRendered;
+            function addRenderedClass() {
+                return isRendered ? fastdom.write(function () {
+                    advert.node.classList.add('ad-slot--rendered');
```

Comment:
This discussion is worth having but unrelated to this PR, so I reverted to using bonzo for DOM manipulations.